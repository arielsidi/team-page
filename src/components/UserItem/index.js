//@flow
import React from "react";
import styled from "styled-components";

import PersonalInfo from "../PersonalInfo";

const HolidaysBar = styled.div`
    background: #ecedf2;
    border-radius: ${props => props.theme.borderRadius};
    width: 150px;
    height: 18px;
    div {
        background: ${props => props.theme.primaryColor};
        height: 100%;
        width: 50%;
        border-top-left-radius: ${props => props.theme.borderRadius};
        border-bottom-left-radius: ${props => props.theme.borderRadius};
    }
`;

const MemberItem = props => {
    // const {personalInfo: }
    return (
        <tr>
            <td>
                <PersonalInfo />
            </td>
            <td>53231</td>
            <td>
                <HolidaysBar>
                    <div />
                </HolidaysBar>
            </td>
            <td>12/02/2018</td>
        </tr>
    );
};

export default MemberItem;
