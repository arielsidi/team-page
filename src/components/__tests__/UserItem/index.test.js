import React from "react";
import UserItem from "../../UserItem";
import { shallow } from "enzyme";

describe("User item", () => {
    const userData = [
        {
            ImageURL: "",
            Acronym: "JR",
            FirstName: "John",
            LastName: "Rambo",
            DepartmentId: 1910,
            JobTitleId: 12332,
            UserKey: 31232,
            UsedDays: 11,
            AvailableDays: 8,
            EmployeeStartDate: "2005-01-03T00:00:00.000"
        },
        {
            ImageURL: "",
            Acronym: "JR",
            FirstName: "John",
            LastName: "Rambo",
            DepartmentId: 1910,
            JobTitleId: null,
            UserKey: 31232,
            UsedDays: 11,
            AvailableDays: 8,
            EmployeeStartDate: "2005-01-03T00:00:00.000"
        }
    ];
    userData.forEach(data => {
        it(`Should render item with data: ${JSON.stringify(data)}.`, () => {
            let wrapper = shallow(<UserItem {...data} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
