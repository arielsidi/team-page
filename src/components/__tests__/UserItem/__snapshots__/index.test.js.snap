// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User item Should render item with data: {"ImageURL":"","Acronym":"JR","FirstName":"John","LastName":"Rambo","DepartmentId":1910,"JobTitleId":12332,"UserKey":31232,"UsedDays":11,"AvailableDays":8,"EmployeeStartDate":"2005-01-03T00:00:00.000"}. 1`] = `
<tr>
  <td>
    <div
      className="PersonalInfo__StyledPersonalInfo-sc-1vtz0dj-0 keMiAq"
    >
      <div
        className="PersonalInfo__UserImage-sc-1vtz0dj-1 joRAcz"
        style={
          Object {
            "backgroundColor": "#999999",
          }
        }
      >
        <span>
          JR
        </span>
      </div>
      <div
        className="PersonalInfo__UserNameAndRol-sc-1vtz0dj-2 ebAIuj"
      >
        John
         
        Rambo
        <span />
      </div>
    </div>
  </td>
  <td>
    31232
  </td>
  <td>
    <div
      className="UserItem__ProgressBarContainer-gqv508-0 frgcqb"
    >
      <div
        className="ProgressBar__StyledProgressBar-sc-1nnf4cm-0 bAMphD"
        title="11/8"
      >
        <div
          style={
            Object {
              "width": "138%",
            }
          }
          title="11/8"
        />
      </div>
    </div>
  </td>
  <td>
    01/03/2005
  </td>
</tr>
`;

exports[`User item Should render item with data: {"ImageURL":"","Acronym":"JR","FirstName":"John","LastName":"Rambo","DepartmentId":1910,"JobTitleId":null,"UserKey":31232,"UsedDays":11,"AvailableDays":8,"EmployeeStartDate":"2005-01-03T00:00:00.000"}. 1`] = `
<tr>
  <td>
    <div
      className="PersonalInfo__StyledPersonalInfo-sc-1vtz0dj-0 keMiAq"
    >
      <div
        className="PersonalInfo__UserImage-sc-1vtz0dj-1 joRAcz"
        style={
          Object {
            "backgroundColor": "#999999",
          }
        }
      >
        <span>
          JR
        </span>
      </div>
      <div
        className="PersonalInfo__UserNameAndRol-sc-1vtz0dj-2 ebAIuj"
      >
        John
         
        Rambo
        <span />
      </div>
    </div>
  </td>
  <td>
    31232
  </td>
  <td>
    <div
      className="UserItem__ProgressBarContainer-gqv508-0 frgcqb"
    >
      <div
        className="ProgressBar__StyledProgressBar-sc-1nnf4cm-0 bAMphD"
        title="11/8"
      >
        <div
          style={
            Object {
              "width": "138%",
            }
          }
          title="11/8"
        />
      </div>
    </div>
  </td>
  <td>
    01/03/2005
  </td>
</tr>
`;
